src/main/scala/li/cil/oc/util/InventoryUtils.scala

```diff
@@ -61,40 +87,31 @@ object InventoryUtils {
    * The number of items inserted can be limited, to avoid unnecessary
    * changes to the inventory the stack may come from, for example.
    */
-  def insertIntoInventorySlot(stack: ItemStack, inventory: IInventory, side: Option[EnumFacing], slot: Int, limit: Int = 64, simulate: Boolean = false) =
+  def insertIntoInventorySlot(stack: ItemStack, inventory: IItemHandler, slot: Int, limit: Int = 64, simulate: Boolean = false): Boolean =
     (stack != null && limit > 0) && {
```

Comment:
Add an `&& stack.stackSize > 0` here, otherwise we'll insert empty stacks into item handlers (e.g. vanilla chests). In particular when called via `insertIntoInventory` due to the loop in there (filling all the slots with empty stacks after the input has been consumed).

This was caught by the same check below before.